pkg/monitoring/vms/prometheus/prometheus.go

```diff
@@ -59,136 +71,105 @@ var (
 	)
 
 	// lower level metrics
-	storageIopsDesc = prometheus.NewDesc(
-		"kubevirt_vmi_storage_iops_total",
-		"I/O operation performed.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "drive", "type",
-		},
-		nil,
-	)
-	storageTrafficDesc = prometheus.NewDesc(
-		"kubevirt_vmi_storage_traffic_bytes_total",
-		"storage traffic.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "drive", "type",
-		},
-		nil,
-	)
-	storageTimesDesc = prometheus.NewDesc(
-		"kubevirt_vmi_storage_times_ms_total",
-		"storage operation time.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "drive", "type",
-		},
-		nil,
-	)
-	vcpuUsageDesc = prometheus.NewDesc(
-		"kubevirt_vmi_vcpu_seconds",
-		"Vcpu elapsed time.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "id", "state",
-		},
-		nil,
-	)
-	networkTrafficBytesDesc = prometheus.NewDesc(
-		"kubevirt_vmi_network_traffic_bytes_total",
-		"network traffic.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "interface", "type",
-		},
-		nil,
-	)
-	networkTrafficPktsDesc = prometheus.NewDesc(
-		"kubevirt_vmi_network_traffic_packets_total",
-		"network traffic.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "interface", "type",
-		},
-		nil,
-	)
-	networkErrorsDesc = prometheus.NewDesc(
-		"kubevirt_vmi_network_errors_total",
-		"network errors.",
-		[]string{
-			"node", "namespace", "name",
-			"domain", "interface", "type",
-		},
+
+	// Metrics descriptors used at the Describe function
+	storageIopsDesc         = prometheus.NewDesc("kubevirt_vmi_storage_iops_total", "", nil, nil)
```

Comment:
Also here. As we spawn goroutine per vmi on node.

Reply:
I see the race condition here, but I'm not sure how to solve this :confused: 

Any suggestions?

Reply:
I know they shouldn't be global variables... but how can I use them at the `Describe` function and also update them at the `update...` functions?

Reply:
Why do you need them to be globally defined if you override them later on anyway per metric which you push?